GetDKAN/dkan
# Add 'hidden' state to dkan_publishing workflow

fixes WCMS-4889

Adding logic that looks at the moderation state and if the moderation state equals publish_without_search_indexing not to index the dataset in the dkan search API. 

## QA Steps

1. Edit an existing dataset or create a new dataset.
  a. Select the 'Publish without search indexing' moderation state at the bottom of the form.
  b. Click Save.
2. Search for the dataset you have 'hidden'.
  a. The dataset should not show up in the search results.
3. Navigate to the hidden dataset.
  a. It should be visible.
4. Unhide & publish the hidden dataset.
5. Search for the dataset.
  a. It should show up in the search results.
